### PR TITLE
chore: Handle generic check destroy in acceptance tests

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
- Added check destroy to almost all objects (check below).
- Changed signatures of ShowByID methods requiring request and not just a plain id
- Removed checks from datasources (just a handful) - current approach is to check the resource under test removal from SF and datasource is not created on the SF side
- Corrected the bad checks that were the results of copy-paste
- Moved existing complex checks with annotation to rework them later

Follow-up needed:
- all the existing CheckDestroy's were addressed but there still may be some tests without CheckDestroy specified; ideally, we should have an automatic way (e.g. as part of architest, to validate this); it will be added to one of the future improvements

Did not work on:
- old *_grant resources and grant_privileges_to_role - deprecated resources
- *_parameter resources - no showbyid nor existing custom check
- snowflake_account_password_policy_attachment - no showbyid nor existing custom check
- snowflake_external_oauth_integration, snowflake_oauth_integration, snowflake_saml_integration, snowflake_scim_integration - no SDK
- snowflake_network_policy_attachment - no showbyid nor existing custom check
- snowflake_table_column_masking_policy_application - no showbyid nor existing custom check
- snowflake_table_constraint - no showbyid nor existing custom check
- snowflake_tag_association - no showbyid nor existing custom check
- snowflake_tag_masking_policy_association no showbyid nor existing custom check
- snowflake_unsafe_execute - a bit different use case (will be handled with the helpers cleanup)
- snowflake_user_public_keys - no showbyid nor existing custom check

The ones that does not have showbyid or existing check will be handled as part of upcoming resource redesign.

Supported objects:
- snowflake_account
- snowflake_alert
- snowflake_api_integration
- snowflake_database
- snowflake_database_role
- snowflake_dynamic_table
- snowflake_email_notification_integration
- snowflake_external_function
- snowflake_external_table
- snowflake_failover_group
- snowflake_file_format
- snowflake_function
- snowflake_managed_account
- snowflake_masking_policy
- snowflake_materialized_view
- snowflake_network_policy
- snowflake_notification_integration
- snowflake_password_policy
- snowflake_pipe
- snowflake_procedure
- snowflake_resource_monitor
- snowflake_role
- snowflake_row_access_policy
- snowflake_schema
- snowflake_sequence
- snowflake_share
- snowflake_stage
- snowflake_storage_integration
- snowflake_stream
- snowflake_table
- snowflake_tag
- snowflake_task
- snowflake_user
- snowflake_view
- snowflake_warehouse

### CODE REVIEW

File: pkg/resources/stage_acceptance_test.go [hunk 1]
```diff
@@ -23,7 +24,7 @@ func TestAcc_StageAlterWhenBothURLAndStorageIntegrationChange(t *testing.T) {
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
 		PreCheck:     func() { acc.TestAccPreCheck(t) },
-		CheckDestroy: nil,
+		CheckDestroy: acc.CheckDestroy(t, resources.Stage),
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/stage_acceptance_test.go:27]: Added.

File: pkg/resources/grant_privileges_to_role_acceptance_test.go [hunk 2]
```diff
@@ -984,10 +984,10 @@ func TestAcc_GrantPrivilegesToRole_OnAllPipes(t *testing.T) {
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: testAccCheckAccountRolePrivilegesRevoked(name),
+		CheckDestroy: acc.CheckAccountRolePrivilegesRevoked(t),
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_role_acceptance_test.go:987]: This is the result of moving the func, we will remove this file soon, so I have not added any other checkdestrroys here (as stated in the PR description). Removing for now.

File: pkg/resources/grant_privileges_to_database_role_acceptance_test.go [hunk 3]
```diff
@@ -1051,7 +1049,7 @@ func TestAcc_GrantPrivilegesToDatabaseRole_RemoveGrantedObjectOutsideTerraform(t
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: testAccCheckDatabaseRolePrivilegesRevoked,
+		CheckDestroy: acc.CheckDatabaseRolePrivilegesRevoked(t),
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/grant_privileges_to_database_role_acceptance_test.go:1052]: added

File: pkg/resources/api_integration_acceptance_test.go [hunk 4]
```diff
@@ -342,7 +337,7 @@ func TestAcc_ApiIntegration_changeApiProvider(t *testing.T) {
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: testAccCheckApiIntegrationDestroy,
+		CheckDestroy: acc.CheckDestroy(t, resources.Account),
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/api_integration_acceptance_test.go:340]: Yes, fixed

File: pkg/datasources/dynamic_tables_acceptance_test.go [hunk 5]
```diff
@@ -42,7 +37,7 @@ func TestAcc_DynamicTables_complete(t *testing.T) {
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: testAccCheckDynamicTableDestroy,
+		CheckDestroy: nil,
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/datasources/dynamic_tables_acceptance_test.go:40]: Nope, it's a datasource

File: pkg/acceptance/check_destroy.go [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+
+var showByIdFunctions = map[resources.Resource]showByIdFunc{
+	resources.Account: func(ctx context.Context, client *sdk.Client, id sdk.ObjectIdentifier) error {
+		return runShowById(ctx, id, client.Accounts.ShowByID)
+	},
+	resources.Alert: func(ctx context.Context, client *sdk.Client, id sdk.ObjectIdentifier) error {
+		return runShowById(ctx, id, client.Alerts.ShowByID)
+	},
+	resources.ApiIntegration: func(ctx context.Context, client *sdk.Client, id sdk.ObjectIdentifier) error {
+		return runShowById(ctx, id, client.Applications.ShowByID)
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/acceptance/check_destroy.go:70]: Yes, fixed

**Review by sfc-gh-jcieslak — DISMISSED** [review]

**Pull request merged**